src/switch_core_io.c

```diff
@@ -81,6 +81,8 @@ SWITCH_DECLARE(switch_status_t) switch_core_session_read_frame(switch_core_sessi
 
 	switch_os_yield();
 
+	switch_log_printf(SWITCH_CHANNEL_SESSION_LOG(session), SWITCH_LOG_DEBUG, "Codec read mutex addr: %p\n", (void *)&session->codec_read_mutex);
```

Comment:
Do you think we can move this when we allocate codec_read_mutex and read_codec and its mutex? So we can set it as an WARNING log.

Reply:
So we also know that in case when it was reset or recreated somehow.

Reply:
I have added logs at the places where mutexes are being created. I also kept this log so we can compare the addresses.